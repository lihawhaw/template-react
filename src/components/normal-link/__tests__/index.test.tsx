import NormalLink from '../index'
import React from 'react'
import renderer from 'react-test-renderer'
import { expect, test } from 'vitest'

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

test('NormalLink changes the class when hovered', () => {
  const component = renderer.create(
    <NormalLink href='https://lihaha.cn'>李哈哈</NormalLink>,
  )
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()

  // manually trigger the callback
  tree.props.onMouseEnter()

  // re-rendering
  tree = toJson(component)
  expect(tree).toMatchSnapshot()

  // manually trigger the callback
  tree.props.onMouseLeave()
  // re-rendering
  tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
