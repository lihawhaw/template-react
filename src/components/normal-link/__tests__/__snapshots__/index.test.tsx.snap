// Vitest Snapshot v1

exports[`NormalLink changes the class when hovered 1`] = `
<a
  className="normal"
  href="https://lihaha.cn"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  李哈哈
</a>
`;

exports[`NormalLink changes the class when hovered 2`] = `
<a
  className="hovered"
  href="https://lihaha.cn"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  李哈哈
</a>
`;

exports[`NormalLink changes the class when hovered 3`] = `
<a
  className="normal"
  href="https://lihaha.cn"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  李哈哈
</a>
`;
